### DISCUSSION TITLE
RAG for External Databases

### DISCUSSION THREAD

**Discussion by snadeem1362** [open-webui/open-webui#3137] [General]
**Is your feature request related to a problem? Please describe.**
I am using open-webui RAG for external documents and it is working perfectly. Now there is a need to connect RAG with our ERP Oracle database. Can you please provide an option to have RAG read the external database to answer the queries?

**Describe the solution you'd like**
An option to connect a database in RAG pipeline. Providing the database as context instead of external documents. Having LLM answering from that database in natural language.

**Describe alternatives you've considered**
Extract data from database and present to LLM.

**Additional context**
As we use RAG for different type of external documents there should be an option to use RAG for different databases to provide dynamic data.

**Comment by tjbck** [comment]
Custom `Tool` would be appropriate in this case!

**Reply by snadeem1362** [reply]
I didnt get you...can you please explain about usage of custom tool?
